controllers/auth.js

```diff
@@ -1,23 +1,70 @@
+const passport = require("passport");
+const User = require("../models/user")
+  
 var getLogin = (req, res) => {
   //TODO: render login page
+  res.render("login", { title: "Login" });
 };
 
 var postLogin = (req, res) => {
   // TODO: authenticate using passport
   //On successful authentication, redirect to next page
+  const user = new User({
+    username: req.body.username,
+    password: req.body.password
+  });
+
+  req.login(user, (err) => {
+    if (err) {
+      console.log(err);
+      res.redirect("/login");
+    } else {
+      res.redirect("/");
+      passport.authenticate("local")(req, res, () => {
+        console.log("authenticated");
+      })
+    }
+  })
 };
 
 var logout = (req, res) => {
   // TODO: write code to logout user and redirect back to the page
+  req.logout();
+  res.redirect("/login");
 };
 
 var getRegister = (req, res) => {
   // TODO: render register page
+  res.render("register", { title: "Register" });
 };
 
 var postRegister = (req, res) => {
```

Comment:
The register function seems to break when I try to register with a user who already has an account.

Reply:
Okay I will look into it and fix this issue.